### DISCUSSION TITLE
Questions regarding synchronization between multiple machines

### DISCUSSION THREAD

**Discussion by xfzv** [twpayne/chezmoi#2432] [Q&A] [answered]
This is not directly related to `chezmoi` but I'm new to Git so I thought I would ask some more experienced users.

My setup:

- A desktop with two OSes (Arch Linux and Gentoo) installed on two distinct SSDs. I can quickly chroot into the other system via `/media/{Arch,Gentoo}`. I have a regular HDD for my data mounted to `/media/Data`.
- A laptop with Arch Linux

I'm currently playing around with `chezmoi` on my desktop and a virtual machine (to experiment with the syncing part). I would rather avoid using an online remote repo (even a private one) on GitHub / GitLab / Bitbucket, so my remote repo would ideally stay locally on my desktop.

For testing purposes, I created the remote repo on the VM and ran  `chezmoi init --ssh vm-username@vm-ip-address:/path/to/remote/repo` on the desktop. It works as expected. I only had an error when trying to `git push` from the desktop but I solved it by running `git config --global receive.denyCurrentBranch updateInstead`.

So, this approach via SSH would most likely work just fine for my laptop with my remote repository located on my desktop, as I intend to do.

However, it wouldn't work for syncing my dotfiles between my Arch Linux and Gentoo installs on my desktop: same machine, same local IP address. I cannot run `git push` to SSH into the same machine with the same local IP address.

Ideally, on my desktop, I would like to be able to manage my dotfiles from the booted OS **and** from chroot. I thought about creating the remote repo somewhere in `/media/Data` but when in chroot, this directory doesn't exist. And it wouldn't even work with SSH in the first place so...

What would you guys suggest? I'm curious to know if there's a way to achieve this or if I would have no other choice but to rely on GitHub / GitLab / Bitbucket for hosting the remote repo.

Any idea is appreciated. Thanks!

**Comment by xfzv** [comment, marked as answer]
I was replying to a comment which partially gave me the solution but looks like it was deleted. To whoever commented: thank you! :+1:

The following did the trick:

```bash
git init --bare /media/Data/path/to/my/local/remote/repo
chezmoi init
chezmoi cd
git remote add origin /media/Data/Software/Linux/dotfiles
```

I also solved the `/media/Data` directory not being accessible from chroot using a bind mount. I had heard about bind mount before but never really got my head around it.

From my `/etc/fstab`:

```
# Original line
/dev/mapper/Data            /media/Data             ext4    defaults,nofail    0 2

# Line I've just added
/media/Data                 /media/Arch/media/Data  none    bind               0 0
```

I'll experiment further and see how it goes.

**Reply by twpayne** [reply]
How did the experimentation go?

**Reply by xfzv** [reply]
> How did the experimentation go?

Very well! I've been using `chezmoi` for a few weeks and this approach works just fine for me.

The bind mount allows me to access `/media/Data/Software/Linux/dotfiles` from both my distros, as expected.
